backend/modules/professor/routers/professor_grading.py

```
@@ -0,0 +1,405 @@
+    except HTTPException:
+        raise
+    except Exception as e:
+        logger.error(f"Error in submit_professor_review: {e!r}", exc_info=True)
+        db.rollback()
+        raise HTTPException(status_code=500, detail=f"Failed to submit review: {e!s}") from e
+
+
+@router.post("/instances/{instance_id}/review/revert", response_model=Dict[str, Any])
+async def revert_to_ai_grade(
+    instance_id: int,
+    current_user: User = Depends(require_professor),
```

Ruff (0.14.8)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)